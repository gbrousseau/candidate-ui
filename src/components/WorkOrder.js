import React, { Component } from 'react';
import Moment from 'moment';

const style = {margin: "50px 50px 0px 100px",width:"auto",textAlign:"left"};

class WorkOrder extends Component {

  constructor(props) {
    super(props);

    this.state = {
      error: null,
      isLoaded: false,
      data: {},
    };

  }

  componentDidMount() {
    fetch("http://localhost:8080/api/workorders/" + this.props.match.params.id)
      .then(res => res.json())
      .then(
        (data) => {
          this.setState({
            isLoaded: true,
            data
          });
        },
        (err) => {
          this.setState({
            isLoaded: true,
            error:err
          });
        }
      )
  }

  render() {
    const { error, isLoaded, data } = this.state;
    if (error) {
      return <div style={style}>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div style={style}>Loading...</div>;
    } else {
      return (
        <div style={style}>
          <div className="col-12 float-left">
            <h2>General</h2>
            <hr />
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Id:</div><div className="d-inline w-50 p-5">{data.id ? data.id : "No data"}</div>
            </div>
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Root id:</div><div className="d-inline w-50 p-5">{data.rootId ? data.rootId : "No data"}</div>
            </div>
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Number:</div><div className="d-inline w-50 p-5">{data.number ? data.number : "No data"}</div>
            </div>
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Title:</div><div className="d-inline w-50 p-5">{data.title ? data.title : "No data"}</div>
            </div>
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Description:</div><div className="d-inline w-50 p-5">{data.description ? data.description : "No data"}</div>
            </div>
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Priority:</div><div className="d-inline w-50 p-5">{data.priority ? data.priority : "No data"}</div>
            </div>
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Status:</div><div className="d-inline w-50 p-5">{data.status ? data.status : "No data"}</div>
            </div>
            <div className="col-4 float-left">
              <div className="w-50 d-inline font-weight-bold">Category:</div><div className="d-inline w-50 p-5">{data.category ? data.category : "No data"}</div>
            </div>
          </div>
          {data.schedule == null ?
            null
            : <div className="col-12 float-left">
              <h2 className="my-3">Schedule</h2>
              <hr />
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Value:</div><div className="d-inline w-50 p-5">{data.schedule.value}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Interval:</div><div className="d-inline w-50 p-5">{data.schedule.interval}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Type:</div><div className="d-inline w-50 p-5">{data.schedule.type}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Days:</div><div className="d-inline w-50 p-5">{data.schedule.days}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Use last completed:</div><div className="d-inline w-50 p-5">{data.schedule.useLastCompleted}</div>
              </div>
              <div className="col-4 float-left" />
            </div>
          }
          {data.createdBy == null ?
            null
          :<div className="col-12 float-left">
              <h2 className="my-3">Created By</h2>
              <hr />
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Id:</div><div className="d-inline w-50 p-5">{data.createdBy.id ? data.createdBy.id : "No data"}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">User name:</div><div className="d-inline w-50 p-5">{data.createdBy.username ? data.createdBy.username : "No data"}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">First name:</div><div className="d-inline w-50 p-5">{data.createdBy.firstName ? data.createdBy.firstName : "No data"}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Last name:</div><div className="d-inline w-50 p-5">{data.createdBy.lastName ? data.createdBy.lastName : "No data"}</div>
              </div>
            </div>
          }
          {data.location == null ?
            null
            :<div className="col-12 float-left">
              <h2 className="my-3">Location</h2>
              <hr />
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">Id:</div><div className="d-inline w-50 p-5">{data.location.id ? data.location.id : "--"}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">User name:</div><div className="d-inline w-50 p-5">{data.location.name ? data.location.name : "--"}</div>
              </div>
              <div className="col-4 float-left">
                <div className="w-50 d-inline font-weight-bold">First name:</div><div className="d-inline w-50 p-5">{data.location.address ? data.location.address : "--"}</div>
              </div>
            </div>
          }
        </div>
      );
    }
  }

}

export default WorkOrder;