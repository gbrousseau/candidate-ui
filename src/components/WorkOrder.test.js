import React from "react";
import { render} from '@testing-library/react';
import { act } from "react-dom/test-utils";
import WorkOrder from "./WorkOrder";

let container = null;

test("renders work order", async () => {
  const fakeWorkOrder = {
    "id": "k3qoOGdq0B",
    "title": "Measure Strap length",
    "description": "Test description",
  };

  jest.spyOn(global, "fetch").mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve(fakeWorkOrder)
    })
  );

  await act(async () => {

    container = render(<WorkOrder match={{ params: { id: 'TEST' } }} />);
  });

  expect(container.getByTestId('id').textContent).toBe(fakeWorkOrder.id);
  expect(container.getByTestId('title').textContent).toBe(fakeWorkOrder.title);
  expect(container.getByTestId('description').textContent).toContain(fakeWorkOrder.description);

  global.fetch.mockRestore();
});