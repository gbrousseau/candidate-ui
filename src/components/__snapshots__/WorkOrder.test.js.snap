// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin: 50px 50px 0px 100px; width: auto; text-align: left;"
      >
        <div
          class="col-12 float-left"
        >
          <h2>
            General
          </h2>
          <hr />
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Id:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
              data-testid="id"
            >
              k3qoOGdq0B
            </div>
          </div>
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Root id:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
            >
              No data
            </div>
          </div>
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Number:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
            >
              No data
            </div>
          </div>
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Title:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
              data-testid="title"
            >
              Measure Strap length
            </div>
          </div>
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Description:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
              data-testid="description"
            >
              Test description
            </div>
          </div>
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Priority:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
            >
              No data
            </div>
          </div>
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Status:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
            >
              No data
            </div>
          </div>
          <div
            class="col-4 float-left"
          >
            <div
              class="min-vw-50 d-inline font-weight-bold"
            >
              Category:
            </div>
            <div
              class="d-inline min-vw-50 p-5"
            >
              No data
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin: 50px 50px 0px 100px; width: auto; text-align: left;"
    >
      <div
        class="col-12 float-left"
      >
        <h2>
          General
        </h2>
        <hr />
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Id:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
            data-testid="id"
          >
            k3qoOGdq0B
          </div>
        </div>
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Root id:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
          >
            No data
          </div>
        </div>
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Number:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
          >
            No data
          </div>
        </div>
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Title:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
            data-testid="title"
          >
            Measure Strap length
          </div>
        </div>
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Description:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
            data-testid="description"
          >
            Test description
          </div>
        </div>
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Priority:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
          >
            No data
          </div>
        </div>
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Status:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
          >
            No data
          </div>
        </div>
        <div
          class="col-4 float-left"
        >
          <div
            class="min-vw-50 d-inline font-weight-bold"
          >
            Category:
          </div>
          <div
            class="d-inline min-vw-50 p-5"
          >
            No data
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
